hail/src/main/scala/is/hail/HailContext.scala

```diff
@@ -437,87 +426,16 @@ object HailContext {
   def pyRemoveIrVector(id: Int) {
     get.irVectors.remove(id)
   }
-}
-
-class HailContext private(
-  val backend: Backend,
-  val logFile: String,
-  val tmpDirPath: String,
-  val branchingFactor: Int,
-  val optimizerIterations: Int) {
-  def stop(): Unit = HailContext.stop()
-
-  def sparkBackend(): SparkBackend = backend.asSpark()
-
-  def sc: SparkContext = sparkBackend().sc
-
-  def fs: FS = sparkBackend().fs
-
-  def fsBc: Broadcast[FS] = sparkBackend().fsBc
-
-  lazy val tmpDir: String = {
-    val tmpDir = TempDir.createTempDir(tmpDirPath, fs)
-    info(s"Hail temporary directory: $tmpDir")
-    tmpDir
-  }
-
-  val flags: HailFeatureFlags = new HailFeatureFlags()
-
-  var checkRVDKeys: Boolean = false
-
-  private var nextVectorId: Int = 0
-  val irVectors: mutable.Map[Int, Array[_ <: BaseIR]] = mutable.Map.empty[Int, Array[_ <: BaseIR]]
-
-  def addIrVector(irArray: Array[_ <: BaseIR]): Int = {
-    val typ = irArray.head.typ
-    irArray.foreach { ir =>
-      if (ir.typ != typ)
-        fatal("all ir vector items must have the same type")
-    }
-    irVectors(nextVectorId) = irArray
-    nextVectorId += 1
-    nextVectorId - 1
-  }
-
-  def version: String = is.hail.HAIL_PRETTY_VERSION
-
-  private[this] def fileAndLineCounts(
-    regex: String,
-    files: Seq[String],
-    maxLines: Int
-  ): Map[String, Array[WithContext[String]]] = {
-    val regexp = regex.r
-    sc.textFilesLines(fs.globAll(files))
-      .filter(line => regexp.findFirstIn(line.value).isDefined)
-      .take(maxLines)
-      .groupBy(_.source.file)
-  }
-
-  def grepPrint(regex: String, files: Seq[String], maxLines: Int) {
-    fileAndLineCounts(regex, files, maxLines).foreach { case (file, lines) =>
-      info(s"$file: ${ lines.length } ${ plural(lines.length, "match", "matches") }:")
-      lines.map(_.value).foreach { line =>
-        val (screen, logged) = line.truncatable().strings
-        log.info("\t" + logged)
-        println(s"\t$screen")
-      }
-    }
-  }
-
-  def grepReturn(regex: String, files: Seq[String], maxLines: Int): Array[(String, Array[String])] =
-    fileAndLineCounts(regex, files, maxLines).mapValues(_.map(_.value)).toArray
-
-  def getTemporaryFile(nChar: Int = 10, prefix: Option[String] = None, suffix: Option[String] = None): String =
-    fs.getTemporaryFile(tmpDir, nChar, prefix, suffix)
 
   def readPartitions[T: ClassTag](
+    fs: FS,
     path: String,
-    partFiles: Array[String],
+    partFiles: IndexedSeq[String],
```

Comment:
does this need to change? you pass in as array below; I know Scala performs implicit conversions, so not a big deal

Reply:
Yes, I changed it for equality.  IndexedSeq implements element-wise equality, while Array implements reference equality.